unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarMovementController.cs

```diff
@@ -60,11 +60,13 @@ public void OnPoolRelease()
 
         void OnWorldReposition(Vector3 current, Vector3 previous) { avatarTransform.position = PositionUtils.WorldToUnityPosition(currentWorldPosition); }
 
-        public void OnTransformChanged(DCLTransform.Model model)
+        public void OnTransformChanged(object model)
```

Comment:
Only for curiosity: why are we now receiving the model as `object` and not typed as `DCLTransform.Model`?

Reply:
When I moved the interfaces to their own assemblies, realized that we can't have specific component references in `IDCLEntity` because we would referencing a high level system from a low level one. This is an anti-pattern as violates 
the architecture layered patterns principle, and dependency inversion principle. 

In this specific example, the design issue would generate cyclic references when trying to extract `DCLTransform` to its own assembly.

The solution is to extract the `OnChanged` events to another system, maybe to the component themselves. There's a card for addressing this: https://github.com/decentraland/unity-renderer/issues/452

Reply:
Understood! Thanks!